[Gavalion] ip

Gavalion opened the pull request:
change the chatbot name from duke to bryan (filename is still duke) create task class for the chatbot
create the checkbox
create setter and getter for description of task

gavalion5 committed:
update code to follow up to week 3 instruction

change the chatbot name from duke to bryan (filename is still duke)
create task class for the chatbot
create the checkbox
create setter and getter for description of task

OmJRamdas commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,69 @@
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
```
Maybe try to separate out the greeting into its own function

OmJRamdas commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,69 @@
+        Scanner userInput = new Scanner(System.in);  // Create a Scanner object
+        System.out.println("how can I help you?");
+        String userString;
+        Task [] taskArray = new Task[100];
```
Create a new class to interface with each Tasks

OmJRamdas commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,69 @@
+        Task [] taskArray = new Task[100];
+        int counter = 0;
+        int numberMark;
+        while (true){
```
add in more comments and break this down into more methods

rchlai replied:
Try adding a space between ) and { in the while statement

OmJRamdas commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,69 @@
+                continue;
+            }
+            if(userString.contains("mark")){
+                numberMark = Integer.parseInt(userString.substring( userString.indexOf(" ")+1));
```
good job on camelCase naming convention

OmJRamdas commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,32 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    protected String taskType;
```
is this a redundant parameter?

rchlai commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,69 @@
+        int numberMark;
+        while (true){
+            userString = userInput.nextLine();
+            if (userString.equals("bye")){
```
Add a space between ) and { in the if statement

rchlai commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,69 @@
+            if (userString.equals("bye")){
+                break;
+            }
+            if(userString.equals("list")){
```
Try adding a space between if and (

rchlai commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,69 @@
+        }
+        sayBye();
+    }
+    public static void printTask(Task[] taskArray){
+        String checkbox;
+        for (int i = 0; i<taskArray.length ;i++){
+            if (taskArray[i] == null){
+                break;
+            }
+            checkbox = "["+ taskArray[i].getStatusIcon() +"]";
+            System.out.println(i+1 + "."+ checkbox+ " "+ taskArray[i].getDescription());
+        }
+    }
+
+    public static void sayBye(){
+        System.out.println("Goodbye, see you again!");
+    }
+    public static void echo(String userString){
+        System.out.println(userString);
+    }
+    public static void printLine(){
+        System.out.println("----------------------------------------------------");
+    }
+    public static void printBryanLogo() {
+        System.out.println("BBBBB  RRRR   Y   Y   A   N   N");
+        System.out.println("B    B R   R   Y Y   A A  NN  N");
+        System.out.println("BBBBB  RRRR     Y   A   A N N N");
+        System.out.println("B    B R  R     Y   AAAAA N  NN");
+        System.out.println("BBBBB  R   R    Y   A   A N   N");
     }
 }
```
Great job packaging functionalities into methods for better readability

rchlai reviewed (commented):
The code is well-designed for better readability. Just have to look out for really minor coding standard violations like omitting spaces.

gavalion5 committed:
Level 4 commit

Added Todo, Deadline and Event
override the toString method

gavalion5 committed:
level 5 commit

add exception and update deadline, todo, event classes

gavalion5 committed:
level 5 with packages and fixes

update the deadline event todo clases that was not comitted previously
add packages

gavalion5 committed:
level 6 commit

change array to arraylist data type, add delete feature

gavalion5 committed:
update code, remove comments

gavalion5 committed:
level-7 commit

add txt file and save feature

gavalion5 committed:
Merge branch 'branch-level-6'

# Conflicts:
#	data1/bryan1.txt
#	src/main/java/bryan/exception/InvalidTextFormatException.java

gavalion5 committed:
Jar file added

gavalion5 committed:
fix delete function

previously delete has a bug where it remove the task from the list and append the rest of task to the txt file instead of removing it

okkhoy commented on src/main/java/bryan/Bryan.java:
```diff
@@ -0,0 +1,313 @@
+package bryan;
+
+import bryan.exception.*;
```
do not use wild-card imports. import the classes as needed

okkhoy commented on src/main/java/bryan/Bryan.java:
```diff
@@ -0,0 +1,313 @@
+    public static final String FILE_PATH = "data1/bryan1.txt"; //change to data/bryan.txt for real test
+
+
+    public static Task stringToTask(String input) throws MarkingException, EmptyEventDescription, InvalidDeadlineFormatException, InvalidEventFormatException, InvalidTodoFormatException, EmptyDeadlineDescription, InvalidTextFormatException {
```
line too long by our coding standards

okkhoy commented on src/main/java/bryan/Bryan.java:
```diff
@@ -0,0 +1,313 @@
+
+    public static void main(String[] args) {
+
+//        System.out.println("full path: " + f.getAbsolutePath());
```
don't retain commented out code

okkhoy commented on src/main/java/bryan/Bryan.java:
```diff
@@ -0,0 +1,313 @@
+    }
+
+
+    public static void main(String[] args) {
```
this method is too long.

okkhoy commented on src/main/java/bryan/Bryan.java:
```diff
@@ -0,0 +1,313 @@
+        int numberMark;
+        Task task;
+        boolean isNumberOfTaskChanged;
+        while (true) {
```
the method is doing too many things.

okkhoy commented on src/main/java/bryan/Bryan.java:
```diff
@@ -0,0 +1,313 @@
+        if (!isValid) {
+            throw new InvalidCommandException();
+        }
+
+
```
inconsistent vertical spacing

okkhoy commented on src/main/java/bryan/exception/EmptyDeadlineDescription.java:
```diff
@@ -0,0 +1,4 @@
+package bryan.exception;
+
+public class EmptyDeadlineDescription extends Exception{
+}
```
pertains to all exception classes. Consider at least having a message that distinguishes the exceptions

okkhoy commented on src/main/java/bryan/Bryan.java:
```diff
@@ -0,0 +1,313 @@
+        System.out.println("BBBBB  R   R    Y   A   A N   N");
+    }
+
+    public static void commandChecker(String userString) throws
```
non trivial methods need header comments

okkhoy reviewed (commented):
@Gavalion the code can be improved quite a lot. 
The main methods needs serious refactoring; it looks like it is doing all the work in the project.
There are too many long lines in the codebase. Please split them appropriately.
None of the non-trivial methods have any javadoc comments

gavalion5 committed:
Added more OOP classes

added TaskList
added Parser
added Ui
added Storage

gavalion5 committed:
add level 9

add find function, add say bye

gavalion5 committed:
add level 8

change date time data to deadline
add after and before feature to filter deadline before and after desired date

gavalion5 committed:
Add Java-doc

add some java documentation to some methods

Gavalion committed:
Merge pull request #1 from Gavalion/branch-level-8

add level 8

Gavalion committed:
Merge branch 'master' into branch-level-9

Gavalion committed:
Merge pull request #2 from Gavalion/branch-level-9

add level 9

Gavalion committed:
Merge branch 'master' into branch-javadoc

Gavalion committed:
Merge pull request #3 from Gavalion/branch-javadoc

Add Java-doc

gavalion5 committed:
Javadoc updated

update java documentation to almost all methods

gavalion5 committed:
user guide

gavalion5 committed:
updated user guide

gavalion5 committed:
JAR realease
